### PR TITLE
Implement concat_extend for BufferVec

### PR DESCRIPTION
**Pull request by breeswish**
###  What have you changed?

This PR implements `concat_extend` for `BufferVec` to be used in https://github.com/tikv/tikv/pull/6132

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Unit test

### CODE REVIEW

File: components/tikv_util/src/buffer_vec.rs [hunk 1]
```diff
@@ -86,6 +86,20 @@ impl BufferVec {
         self.data.extend_from_slice(buffer.as_ref());
     }
 
+    /// Appends multiple buffer parts together as one buffer to the back.
+    #[inline]
+    pub fn push_together<T, I>(&mut self, buffer_parts: I)
```
Comment (siddontang) [inline components/tikv_util/src/buffer_vec.rs]: @nrc 

which better name do you suggest? I think `push_together` is strange.

Comment (BusyJay) [reply components/tikv_util/src/buffer_vec.rs]: You can implement the `std::iter::Extend` trait instead.

Comment (breeswish) [reply components/tikv_util/src/buffer_vec.rs]: The `Extend` trait is already implemented for the non-together version, i.e. extending 5 `&[u8]`s results in 5 new elements. @BusyJay

Comment (BusyJay) [reply components/tikv_util/src/buffer_vec.rs]: How about `concat_extend`?

Comment (breeswish) [reply components/tikv_util/src/buffer_vec.rs]: The naming LGTM.

Comment (nrc) [reply components/tikv_util/src/buffer_vec.rs]: I agree `concat_extend` is good. Other alternatives: `collect_and_append` (from `Iterator::collect`), `flat_append` or `flat_push` (from `Iterator::flat_map`)

File: components/tikv_util/src/buffer_vec.rs [hunk 2]
```diff
@@ -598,8 +634,18 @@ mod tests {
...
+        v3.extend(&[0xAA, 0xAB, 0xCC]);
+        assert_eq!(v3.len(), 5);
+        assert_eq!(v3.total_len(), 6);
+        assert_eq!(format!("{:?}", v3), "[0C00, null, null, AA, AAABCC]");
+
+        v3.extend(&[]);
+        assert_eq!(v3.len(), 6);
+        assert_eq!(v3.total_len(), 6);
+        assert_eq!(format!("{:?}", v3), "[0C00, null, null, AA, AAABCC, null]");
+
```
Comment (mapleFU) [inline components/tikv_util/src/buffer_vec.rs]: I found there are implemention for `Extend<&'a u8>`, is it proper to adding testing like:
```
v3.extend(vec![0xAA]);
...
```
?

**Pull request merged**